Added SvgForTheBadgeRenderer

ricardoboss opened the pull request:
Hi! In another project using this library, I requested to add support for more styles: [PUGX/badge-poser#705](https://github.com/PUGX/badge-poser/issues/705). (PR to fix that one: https://github.com/PUGX/badge-poser/pull/707)

All styles were already there, except from `for-the-badge`. So I implemented it in this PR. This is what the new style looks like:

![image](https://user-images.githubusercontent.com/6266356/151266025-bdefa5a3-5ed5-45c1-930c-1b9a2d4b9eb9.png)

Note: there is a new dependency on [kartsims/easysvg](https://github.com/kartsims/easysvg). The only problem this library has, is that it still supports PHP 5.6 (oof). Maybe no one bothered to update it or the author wants to support older PHP version. Anyhow, I [created a fork of it](https://github.com/ricardoboss/easysvg), which I added as a VCS repository in `composer.json`. That way, this PR doesn't have to wait until my PR over there is merged: https://github.com/kartsims/easysvg/pull/32

Let me know what you think!

JellyBellyDev commented:
Hi @ricardoboss! Thanks for your PR! Can you fix the CI, please?

ricardoboss commented:
@JellyBellyDev fixed (I hope)

antonkomarev commented on composer.json:
```diff
@@ -29,12 +29,17 @@
+        },
+        {
+            "type": "vcs",
+            "url": "https://github.com/ricardoboss/easysvg"
```
do we really need to depend on the fork of the third party lib?

ricardoboss replied:
What do you mean by unpublished?

I specifically explained why this fork has a release: https://github.com/ricardoboss/easysvg/releases/tag/2.0

We need this fork, since the original library has a lot of deprecation warnings (and even fatal errors) because it supported very old PHP versions. In my fork, I fixed them and opened a PR. I am in talks with the creator of the library to merge my PR and then we can drop this extra repository.

JellyBellyDev replied:
Nice! I pinged https://github.com/kartsims/easysvg/pull/32#issuecomment-1038998987 :)

antonkomarev commented:
Why Roboto and Montserrat fonts are used?
[Original style](https://img.shields.io/badge/some_example-text_here-red?style=for-the-badge) `for-the-badge` just uses common `Verdana,Geneva,DejaVu Sans,sans-serif`.
<img width="742" alt="Screenshot 2022-01-27 at 17 47 58" src="https://user-images.githubusercontent.com/1849174/151382303-6645f134-7f23-4f9a-b333-3723c7da9975.png">

antonkomarev commented on composer.json:
```diff
@@ -29,12 +29,17 @@
         "php": ">=7.4",
         "ext-gd": "*",
         "ext-simplexml": "*",
+        "kartsims/easysvg": "^2.0",
```
Do we really need extra dependency? How svg rendering is implemented in other badge styles without it?

ricardoboss replied:
In https://github.com/ekfuhrmann/badge-generator it is implemented using opentype.js. We need some way to render a font in SVG, since not everyone has the required fonts installed on their systems. If you another way, let me know!

JellyBellyDev replied:
No problem for me to new deps but i prefer to use the standard font as talk by @antonkomarev here: https://github.com/badges/poser/pull/86#issuecomment-1023287355

ricardoboss commented:
> @ricardoboss thanks for pushing this forward!
> 
> Why Roboto and Montserrat fonts are used? [Original style](https://img.shields.io/badge/some_example-text_here-red?style=for-the-badge) `for-the-badge` just uses common `Verdana,Geneva,DejaVu Sans,sans-serif`.
> 

I used the same fonts that are used by the original "for the badge" generator: https://github.com/ekfuhrmann/badge-generator
If they're wrong, they can easily be replaced.

> Output image looks blurred and not flat (it has some kind of bevel border on the top).
> 
> > ![image](https://user-images.githubusercontent.com/6266356/151266025-bdefa5a3-5ed5-45c1-930c-1b9a2d4b9eb9.png)

The bevel on top is just my poor screenshot of the badge :) The svg itself is flag.

JellyBellyDev reviewed (changes requested):
Yoy can add to readme how to use the new feature?
Thanks

garak commented on README.md:
````diff
@@ -65,7 +70,7 @@ $image = $poser->generate('license', 'MIT', '428F7E', 'plastic');
 ```
 
-The allowed styles are: `plastic`, `flat` and `flat-square`.
+The allowed styles are: `plastic`, `flat`, `flat-square` and `for-the-badge`.
````
We miss a comma before "and" (I know, we missed it even before, but better late than never)

ricardoboss commented:
@antonkomarev I addressed your comments in the last commits. I took a look at the shields.io `for-the-badge` badges and adjusted the fonts and layout:

![test](https://user-images.githubusercontent.com/6266356/153877823-2c850d36-ac67-403d-abfd-23b7e958bdca.svg)

JellyBellyDev reviewed (approved):
Thanks @ricardoboss!
For now i merged but i open a issue to remove the unpublished fork deps as soon as the PR is released
